lib/SSH/LibSSH.pm6

```diff
@@ -795,6 +795,15 @@ class SSH::LibSSH {
                         !! StreamingDecoder.new(Rakudo::Internals.NORMALIZE_ENCODING(
                                 $enc // 'utf-8'));
                     $loop.add-poller: -> $remove is rw {
+                        if ssh_channel_is_eof($!channel-handle) {
+                            $remove = True;
+                            unless $bin {
+                                $s.emit($decoder.consume-all-chars());
+                            }
+                            $s.done();
+                            return;
```

Comment:
This `return` is not going to behave well. `return` binds to the lexically enclosing `sub` or `method`, not a block. It only "works" in so far as it results in an exception, which in turn is handled by the `CATCH` below, which will do a `quit`, which will be dropped because `done` was already called and a stream can only end in one way.  I suggest removing the return and moving what follows into an `else`.

Reply:
Thanks! Updated